src/main/java/ttt/Board.java

```diff
@@ -1,94 +1,116 @@
 package ttt;
 
 import java.util.Arrays;
+import java.util.function.Function;
 
 import static ttt.PlayerSymbol.*;
-import static ttt.PlayerSymbol.VACANT;
 
 public class Board {
-    private PlayerSymbol[] grid = new PlayerSymbol[BOARD_DIMENSION * BOARD_DIMENSION];
-
     protected static final int BOARD_DIMENSION = 3;
     private static final int NUMBER_OF_SLOTS = BOARD_DIMENSION * BOARD_DIMENSION;
+    private Cell[] grid = new Cell[BOARD_DIMENSION * BOARD_DIMENSION];
 
     public Board() {
-        for (int i = 0; i < NUMBER_OF_SLOTS; i++) {
-            grid[i] = VACANT;
+        for (int cellIndex = 0; cellIndex < NUMBER_OF_SLOTS; cellIndex++) {
+            grid[cellIndex] = new Cell(calculateOffsetFor(cellIndex), VACANT);
         }
     }
 
-    public Board(PlayerSymbol... grid) {
-        this.grid = grid;
+    public Board(PlayerSymbol... initialGridLayout) {
+        for (int cellIndex = 0; cellIndex < NUMBER_OF_SLOTS; cellIndex++) {
+            this.grid[cellIndex] = new Cell(calculateOffsetFor(cellIndex), initialGridLayout[cellIndex]);
+        }
+    }
+
+    public Cell[][] getRows() {
+        LineGenerator lines = new LineGenerator(grid);
+        return new Cell[][]{lines.topRow(), lines.middleRow(), lines.bottomRow()};
     }
 
     public boolean hasWinningCombination() {
-        return hasWinningRow() || hasWinningColumn() || hasWinningDiagonal();
+        LineGenerator lineGenerator = new LineGenerator(grid);
+        return checkForWinIn(lineGenerator.linesForAllDirections());
+    }
+
+    public PlayerSymbol getWinningSymbol() {
+        if (hasWinningCombination()) {
+
+            for (Cell[] row : getLinesFromGrid()) {
+                if (containsOnly(X, row)) {
+                    return X;
+                }
+                if (containsOnly(O, row)) {
+                    return O;
+                }
+            }
+        }
```

Comment:
A part of me is OK with this, but another one tells me that there is a lot of complexity going on over here.

Somehow is seems like there is something missing. In an ideal world I would love to be able to do something like `cells.winningRow().getSymbol()`